perception/sub8_perception/src/sub8_perception/start_gate.cc

```diff
@@ -0,0 +1,191 @@
+#include <sub8_perception/start_gate.hpp>
+Sub8StartGateDetector::Sub8StartGateDetector() : image_transport(nh), gate_line_buffer(30), running(true) {
+    image_sub = image_transport.subscribeCamera("stereo/right/image_raw", 1,
```

Comment:
You should really look into using ROSCameraStream, it makes this trivial and handles a lot of the boilerplate for you and provides you with an image buffer in case you need temporal information. It also contains all the camera info stuff you need.

You can use it [like this](https://github.com/uf-mil/mil_common/blob/great_merge/perception/mil_vision/nodes/camera_stream_demo.cpp): `mil::ROSCameraStream<cv::Vec3b> left_cam_stream(nh, history_length);` @